firestore/query.go

```diff
@@ -61,6 +62,149 @@ type Query struct {
 	// readOptions specifies constraints for reading results from the query
 	// e.g. read time
 	readSettings *readSettings
+
+	runQuerySettings *runQuerySettings
+}
+
+type runQuerySettings struct {
+	// Explain options for the query. If set, additional query
+	// statistics will be returned. If not, only query results will be returned.
+	explainOptions *pb.ExplainOptions
+}
+
+// newRunQuerySettings creates a runQuerySettings with a given RunOption slice.
+func newRunQuerySettings(opts []RunOption) (*runQuerySettings, error) {
+	s := &runQuerySettings{}
+	for _, o := range opts {
+		if o == nil {
+			return nil, errors.New("firestore: RunOption cannot be nil")
+		}
+		err := o.apply(s)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return s, nil
+}
+
+// ExplainOptions is explain options for the query.
+//
+// Query Explain feature is still in preview and not yet publicly available.
+// Pre-GA features might have limited support and can change at any time.
+type ExplainOptions struct {
+	// When false (the default), the query will be planned, returning only
+	// metrics from the planning stages.
+	// When true, the query will be planned and executed, returning the full
+	// query results along with both planning and execution stage metrics.
+	Analyze bool
+}
+
+func (e ExplainOptions) apply(s *runQuerySettings) error {
+	if s.explainOptions != nil {
+		return errors.New("firestore: ExplainOptions can be specified only once")
+	}
+	pbExplainOptions := pb.ExplainOptions{
+		Analyze: e.Analyze,
+	}
+	s.explainOptions = &pbExplainOptions
+	return nil
+}
```

Comment:
It seems like for this codebase all the options style code is either put in `client.go` (in the case of `readSettings`/`ReadOption` or in `options.go`. Would it be more preferable to put this code in one of those files?

Reply:
Moved to options.go